.github/workflows/rust.yml

```diff
@@ -29,3 +29,16 @@ jobs:
       run: cargo test --verbose --no-default-features
     - name: Run Clippy lints (no default features)
       run: cargo clippy --verbose --no-default-features -- -D warnings
+
+  miri:
+    name: "Miri"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install Miri
+        run: |
+          rustup toolchain install nightly --component miri
+          rustup override set nightly
+          cargo miri setup
+      - name: Test with Miri tests with "from_buffer" in name
+        run: cargo miri test from_buffer
```

Comment:
Might be worth running Miri for all tests, just in case. Or is there a specific reason not to?

Reply:
Yes, unfortunately Miri errors out on many of the tests and doctests. For example `TestDir::new()` depends on `libc::clock_gettime()` which gives the error "`clock_gettime` not available when isolation is enabled". Miri cannot do FFI calls either, so it errors on anything calling `ffi::inotify_init()`: "can't call foreign function: inotify_init".

Reply:
Ah, got it. Too bad, what can we do!